src/lib.rs

```diff
@@ -13,6 +15,14 @@ mod target;
 pub use region::Region;
 pub use target::Target;
 
+// oldest entry available for 2018-05-10 23:30:00
+static OLDEST_VALID_DATE: LazyLock<NaiveDateTime> = LazyLock::new(|| {
```

Comment:
It wouldn't be the end of the world to create a new date - this is called probably max twice but fine also to have a static. I don't get why we need a LazyLock though? Is concurrency an issue?

Reply:
If you try to remove the LazyLock and make `OLDEST_VALID_DATE` just a plain `NaiveDateTime`, e.g.

```Rust
static OLDEST_VALID_DATE: NaiveDateTime = NaiveDate::from_ymd_opt(2018, 5, 10)
    .unwrap()
    .and_hms_opt(23, 30, 0)
    .unwrap();
```

unfortunately we get an error because of the unwrap:

```
$ cargo check
[...]

error: `std::option::Option::<T>::unwrap` is not yet stable as a const fn
  --> src/lib.rs:19:43
   |
19 |   static OLDEST_VALID_DATE: NaiveDateTime = NaiveDate::from_ymd_opt(2018, 5, 10)
   |  ___________________________________________^
20 | |     .unwrap()
   | |_____________^

[...]
```

It's not a concurrency issue per se, although I suspect given we spawn multiple tokio threads it's probably best for this to be thread-safe.

Anyway, that's why the `LazyLock` is useful.